src/chainstate/stacks/db/transactions.rs

```diff
@@ -426,41 +444,56 @@ impl StacksChainState {
-
+                        error!("Runtime error in contract analysis for {}: {}", &contract_id, &e);
+                        let receipt = StacksTransactionReceipt {
+                            transaction: tx.clone(),
+                            events: vec![],
+                            result: Value::err_uint(0),
+                            stx_burned: 0
+                        };
+                
                         // abort now -- no burns
-                        return Ok(0);
+                        return Ok(receipt);
                     }
                 };
 
                 // execution -- if this fails due to a runtime error, then the transaction is still
                 // accepted, but the contract does not materialize (but the sender is out their fee).
-                let asset_map = match clarity_tx.connection().initialize_smart_contract(
+                let (asset_map, events) = match clarity_tx.connection().initialize_smart_contract(
                     &contract_id, &contract_ast, &contract_code_str,
                     |asset_map, _| { !StacksChainState::check_transaction_postconditions(&tx.post_conditions, &tx.post_condition_mode, origin_account, asset_map) }) {
-                    Ok(asset_map) => {
-                        Ok(asset_map)
+                    Ok((asset_map, events)) => {
+                        Ok((asset_map, events))
                     },
                     Err(e) => {
                         match e {
                             // runtime errors are okay -- we just have an empty asset map
                             clarity_error::Interpreter(InterpreterError::Runtime(ref runtime_error, ref stack)) => {
-                                debug!("Runtime error {:?} on instantiating {:?}, code {:?}, stack trace {:?}", runtime_error, &contract_id, &contract_code_str, stack);
-                                Ok(AssetMap::new())
+                                error!("Runtime error {:?} on instantiating {}, code {:?}, stack trace {:?}", runtime_error, &contract_id, &contract_code_str, stack);
+                                Ok((AssetMap::new(), vec![]))
                             },
                             _ => Err(e)
                         }
                     }
                 }.map_err(|e| {
-                    warn!("Invalid smart-contract transaction {}: {:?}", &tx.txid(), &e);
+                    error!("Invalid smart-contract transaction {}: {}", &tx.txid(), &e);
                     Error::ClarityError(e)
                 })?;
                 
                 // store analysis -- if this fails, then the have some pretty bad problems
                 clarity_tx.connection().save_analysis(&contract_id, &contract_analysis)
                     .expect("FATAL: failed to store contract analysis");
                 
-                Ok(asset_map.get_stx_burned_total())
+                    println!("EVENTS: {:?}", events);
```

Comment:
Stray `println!` should be removed.  Use one of the logging macros, please.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1319/commits/868191d59d514ce9aac6ff55da764a9cfefe8aca